apollo-router/src/plugins/telemetry/mod.rs

```diff
@@ -428,42 +446,64 @@ impl Telemetry {
                         EnvFilter::try_new(log_level)
                             .context("could not parse log configuration")?,
                     )
-                    .with_file(true)
-                    .with_line_number(true);
+                    .with_file(
+                        config
+                            .logging
+                            .as_ref()
+                            .map(|l| l.display_filename)
+                            .unwrap_or(default_display_filename()),
+                    )
+                    .with_line_number(
+                        config
+                            .logging
+                            .as_ref()
+                            .map(|l| l.display_line_number)
+                            .unwrap_or(default_display_line_number()),
+                    );
 
                 if let Some(sub) = subscriber {
                     let telemetry = tracing_opentelemetry::layer().with_tracer(tracer);
                     let subscriber = sub.with(telemetry);
                     if let Err(e) = set_global_default(subscriber) {
                         ::tracing::error!("cannot set global subscriber: {:?}", e);
                     }
-                } else if atty::is(atty::Stream::Stdout) {
-                    let telemetry = tracing_opentelemetry::layer().with_tracer(tracer);
-
-                    let subscriber = sub_builder
-                        .event_format(formatters::TextFormatter::new())
-                        .finish()
-                        .with(telemetry);
-                    if let Err(e) = set_global_default(subscriber) {
-                        ::tracing::error!("cannot set global subscriber: {:?}", e);
-                    }
                 } else {
-                    let telemetry = tracing_opentelemetry::layer().with_tracer(tracer);
-
-                    let subscriber = sub_builder
-                        .map_event_format(|e| {
-                            e.json()
-                                .with_current_span(true)
-                                .with_span_list(true)
-                                .flatten_event(true)
-                        })
-                        .map_fmt_fields(|_f| JsonFields::new())
-                        .finish()
-                        .with(telemetry);
-                    if let Err(e) = set_global_default(subscriber) {
-                        ::tracing::error!("cannot set global subscriber: {:?}", e);
+                    match config
+                        .logging
+                        .as_ref()
+                        .map(|l| l.format)
+                        .unwrap_or_default()
```

Comment:
Will this result in a change in behaviour? I'm not sure what the default format is, but I wonder if that will match up with the existing tty detecting behaviour that we have.
If it is a change, we need to discuss that and maybe mark it as breaking change in the changelog.

Reply:
It won't change the default behavior thanks to this impl https://github.com/apollographql/router/pull/2040/files/6dc13330f5e96e475ad3f0b748a0201de2afad69#diff-eb7847da8c5109309df2eacc8f7acd6c73d93260c81890258b37b9dac3f30f7cR239